pkg/integrations/v2/apache_exporter/apache_exporter.go

```diff
@@ -0,0 +1,110 @@
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// DefaultConfig holds the default settings for the apache_exporter integration
+var DefaultConfig = Config{
+	ApacheAddr:         "http://localhost/server-status?auto",
+	ApacheHostOverride: "",
+	ApacheInsecure:     false,
+}
+
+// Config controls the apache_exporter integration.
+type Config struct {
+	ApacheAddr         string               `yaml:"scrape_uri,omitempty"`
+	ApacheHostOverride string               `yaml:"host_override,omitempty"`
+	ApacheInsecure     bool                 `yaml:"insecure,omitempty"`
+	Common             common.MetricsConfig `yaml:",inline"`
+}
+
+// ApplyDefaults applies the integration's default configuration.
+func (c *Config) ApplyDefaults(globals integrations_v2.Globals) error {
+	c.Common.ApplyDefaults(globals.SubsystemOpts.Metrics.Autoscrape)
+	return nil
+}
+
+// Identifier returns a string that identifies the integration.
+func (c *Config) Identifier(globals integrations_v2.Globals) (string, error) {
+	return *c.Common.InstanceKey, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// Name returns the name of the integration this config is for.
+func (c *Config) Name() string {
+	return "apache"
+}
+
+// InstanceKey returns the addr of the apache server.
+func (c *Config) InstanceKey(agentKey string) (string, error) {
+	u, err := url.Parse(c.ApacheAddr)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s:%s", u.Hostname(), u.Port()), nil
+}
+
+type apacheHandler struct {
+	cfg *Config
+	log log.Logger
+}
+
+// NewIntegration instantiates a new integrations.MetricsIntegration
+// which will handle requests to the apache exporter.
+func (c *Config) NewIntegration(logger log.Logger, globals integrations_v2.Globals) (integrations_v2.Integration, error) {
+	var metricsCfg common.MetricsConfig
+	metricsCfg.ApplyDefaults(globals.SubsystemOpts.Metrics.Autoscrape)
+
+	ah := &apacheHandler{cfg: c, log: logger}
+	h, err := ah.createHandler()
+	if err != nil {
+		return nil, err
+	}
+
+	return metricsutils.NewMetricsHandlerIntegration(logger, c, metricsCfg, globals, h)
+}
+
+func (ah *apacheHandler) createHandler() (http.HandlerFunc, error) {
+	_, err := url.ParseRequestURI(ah.cfg.ApacheAddr)
+	if err != nil {
+		level.Error(ah.log).Log("msg", "scrape_uri is invalid", "err", err)
+		return nil, err
+	}
+
+	aeExporter := ae.NewExporter(ah.log, &ae.Config{
+		ScrapeURI:    ah.cfg.ApacheAddr,
+		HostOverride: ah.cfg.ApacheHostOverride,
+		Insecure:     ah.cfg.ApacheInsecure,
+	})
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		registry := prometheus.NewRegistry()
+		registry.MustRegister(aeExporter)
+		h := promhttp.HandlerFor(registry, promhttp.HandlerOpts{})
+		h.ServeHTTP(w, r)
+	}, nil
+}
+
+func init() {
```

Comment:
Nitpicky comment, makes no difference but wdyt about moving init() to the top, for consistency with other integrations?

Reply:
Some other integrations have it at the bottom as it seems.